Unreal/Plugins/AirSim/Source/PIPCamera.cpp

```diff
@@ -71,8 +71,19 @@ void APIPCamera::BeginPlay()
     camera_type_enabled_.assign(imageTypeCount(), false);
 
     for (unsigned int image_type = 0; image_type < imageTypeCount(); ++image_type) {
-        //use final color for all calculations
-        captures_[image_type]->CaptureSource = ESceneCaptureSource::SCS_FinalColorLDR;
+        switch (Utils::toEnum<ImageType>(image_type))
+        {
+        // Capture depth data for depth formats.
+        case ImageType::DepthVis:
+        case ImageType::DepthPerspective:
+        case ImageType::DepthPlanner:
+            captures_[image_type]->CaptureSource = ESceneCaptureSource::SCS_SceneDepth;
```

Comment:
This seems to break the subwindow rendering, only getting a blank red image
Is it absolutely required?

Reply:
Oh oops, thanks for that catch - I didn't try it with the sub windows. I am guessing it is an issue with just DepthVis. Since the value is in centimeters, the red component of DepthVis will almost always be greater than one, which is probably why you see the blank red image. It might work just by removing `case ImageType::DepthVis:`, but I'm not sure yet. I'll try it out later today and get back to you.

Reply:
I made a mistake here, I was confused about how depth was done before. I thought it should be accessing the depth buffer directly, but now I realize that is using a custom post processing material. I will have a fix soon hopefully.

Reply:
Okay, it should be fixed now. The main thing that needed to be changed was just using Float16 for the RGBA.